src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -237,6 +243,26 @@ static void checkAttributes(AnonymousFunctionExpressionSyntax syntax, SyntaxList
             }
         }
 
+        private (RefKind, TypeWithAnnotations) BindExplicitLambdaReturnType(TypeSyntax syntax, BindingDiagnosticBag diagnostics)
+        {
+            MessageID.IDS_FeatureLambdaReturnType.CheckFeatureAvailability(diagnostics, syntax);
+
+            syntax = syntax.SkipRef(out RefKind refKind);
+            var returnType = BindType(syntax, diagnostics);
+            var type = returnType.Type;
+
+            if (returnType.IsStatic)
+            {
+                diagnostics.Add(ErrorFacts.GetStaticClassReturnCode(useWarning: false), syntax.Location, type);
+            }
+            else if (returnType.IsRestrictedType(ignoreSpanLikeTypes: true))
```

Comment:
I think this is missing a varargs check: [https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Source/SourceOrdinaryMethodSymbolBase.cs,610](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Source/SourceOrdinaryMethodSymbolBase.cs,610)